packages/widget/src/web-component.tsx

```diff
@@ -60,14 +69,8 @@ initializeSkipWidget();
 
 export default WebComponent;
 
-type Stringify<T> = {
-  [K in keyof T]?: string;
-};
-
 declare global {
-  namespace JSX {
-    interface IntrinsicElements {
-      [WEB_COMPONENT_NAME]: Stringify<WebComponentProps>;
-    }
+  interface HTMLElementTagNameMap {
+    [WEB_COMPONENT_NAME]: WidgetProps;
```

Comment:
super simple, now the element type is equivalent to `WidgetProps`, so the ide will only show widgetProps in the autocomplete:

![image](https://github.com/user-attachments/assets/29759997-c246-4929-bd36-8573c4826bfb)